apps/web-giddh/src/app/invoice/preview/models/download-voucher/download-voucher.component.ts

```diff
@@ -92,10 +92,12 @@ export class DownloadVoucherComponent implements OnInit, OnDestroy {
                         if (fileType === "base64") {
                             saveAs((this.generalService.base64ToBlob(response.body.attachments[0].encodedData, '', 512)), response.body.attachments[0].name);
                         } else {
-                            saveAs(response, `${this.selectedItem.voucherNumber}.` + 'zip');
+                            let voucherNumber = (this.selectedItem?.voucherNumber) ? this.selectedItem?.voucherNumber : this.commonLocaleData?.app_not_available;
```

Comment:
Consider moving the line:
`let voucherNumber = (this.selectedItem?.voucherNumber) ? this.selectedItem?.voucherNumber : this.commonLocaleData?. app_not_available`
higher in the code so that it needs to ba calculated only once.